server/imap/connection.coffee

```diff
@@ -204,6 +203,13 @@ Imap::multicopy = (uid, paths, callback) ->
         @copy uid, path, cb
     , callback
 
+# remove message from multiple boxes
+Imap::multiremove = (paths, callback) ->
+    async.eachSeries paths, ({path, uid}, cb) =>
+        @deleteMessageInBox path, uid, cb
```

Comment:
It is possible to remove multiple message from a mailbox, but not as we are doing here a message that is in multiple mailboxes.